source/windows/secure_channel_tls_handler.c

```diff
@@ -200,34 +206,84 @@ static int s_manually_verify_peer_cert(struct aws_channel_handler *handler) {
     engine_config.cbSize = sizeof(engine_config);
     engine_config.hExclusiveRoot = sc_handler->custom_ca_store;
 
-    HCERTCHAINENGINE engine = NULL;
     if (!CertCreateCertificateChainEngine(&engine_config, &engine)) {
         AWS_LOGF_ERROR(
             AWS_LS_IO_TLS,
             "id=%p: failed to load a certificate chain engine with SECURITY_STATUS %d. "
             "Most likely, the configured CA is corrupted.",
             (void *)handler,
             (int)status);
-        CertFreeCertificateContext(peer_certificate);
-        return AWS_OP_ERR;
+        goto done;

```

Comment:
There are LOTS of places in this function where we return AWS_OP_ERR, but it doesn't seem like any aws_raise_error() has happened

Reply:
Nevermind. The 2 places that call this function will call `aws_raise_error(AWS_IO_TLS_ERROR_NEGOTIATION_FAILURE);` if this fails. So it's weird, but it's not a bug